README.md

```diff
@@ -9,7 +9,55 @@ GraphQL Schema Bindings is a flexible and scalable GraphQL schema generator that
 
 GraphQL Schema Bindings uses decorators to define the necessary metadata to map your native types to GraphQL types.
 
-Use `@type` to mark a class to be exported to GraphQL.
+Decorators make it easy to create GraphQL services.  Let's start with a simple example.  This example can get users from the GitHub REST API and represent them as GraphQL.
+
+We'll define a simple user GraphQL query.  The first step is to create our user class:
+
+```javascript
+@type
+class User {
+  @context
+  context;
+
+  @field(ID)
+  get id() {
+    return this.data.id;
+  }
+  
+  @field(String)
+  get name() {
+    return this.data.name;
+  }
+
+  constructor(data) {
+    this.data = data;
+  }
+}
+```
+
+Now we can build a query that returns that user:
+
+```javascript
+@type
+class UserQuery {
+  @field(User)
+  async user(@arg(ID) id) {
+    const url = id ? `/${id}` : "";
+    const { data } = await axios.get(`https://api.github.com/users${url}`);
+    return new User(data);
+  }
+}
+```
+
+And that's it.  Run this sample yourself here:  [GitHub users sample](examples/github_users)
+
+# Understanding schema types
+
+Schema types work like decorators to help your functions.  They all have the format of `@<some type>` before a function in your code.
+
+## @type
+
+Use `@type` to mark a class to be exported to GraphQL.  This indicates that a class represents data that will be either returned from or used as an input to GraphQL.
```

Comment:
Input types are marked with the `@input` decorator `@type` is reserved for output types.